packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -211,445 +192,285 @@ void createSettingsAarGradle(Directory androidDirectory) {
+    command.add('-Pshrink=true');
   }
-  throwToolExit('Unsuported Android Plugin version: $androidPluginVersion.');
-  return '';
-}
-
-final RegExp _androidPluginRegExp = RegExp('com\.android\.tools\.build\:gradle\:(\\d+\.\\d+\.\\d+\)');
-
-/// Returns the Gradle version that the current Android plugin depends on when found,
-/// otherwise it returns a default version.
-///
-/// The Android plugin version is specified in the [build.gradle] file within
-/// the project's Android directory.
-String getGradleVersionForAndroidPlugin(Directory directory) {
-  final File buildFile = directory.childFile('build.gradle');
-  if (!buildFile.existsSync()) {
-    return defaultGradleVersion;
+  if (androidBuildInfo.targetArchs.isNotEmpty) {
+    final String targetPlatforms = androidBuildInfo
+      .targetArchs
+      .map(getPlatformNameForAndroidArch).join(',');
+    command.add('-Ptarget-platform=$targetPlatforms');
   }
-  final String buildFileContent = buildFile.readAsStringSync();
-  final Iterable<Match> pluginMatches = _androidPluginRegExp.allMatches(buildFileContent);
-
-  if (pluginMatches.isEmpty) {
-    return defaultGradleVersion;
+  if (shouldBuildPluginAsAar) {
+    // Pass a system flag instead of a project flag, so this flag can be
+    // read from include_flutter.groovy.
+    command.add('-Dbuild-plugins-as-aars=true');
+    // Don't use settings.gradle from the current project since it includes the plugins as subprojects.
+    command.add('--settings-file=settings_aar.gradle');
   }
-  final String androidPluginVersion = pluginMatches.first.group(1);
-  return getGradleVersionFor(androidPluginVersion);
-}
+  command.add(assembleTask);
 
-/// Overwrite local.properties in the specified Flutter project's Android
-/// sub-project, if needed.
-///
-/// If [requireAndroidSdk] is true (the default) and no Android SDK is found,
-/// this will fail with a [ToolExit].
-void updateLocalProperties({
-  @required FlutterProject project,
-  BuildInfo buildInfo,
-  bool requireAndroidSdk = true,
-}) {
-  if (requireAndroidSdk) {
-    _exitIfNoAndroidSdk();
+  final Stopwatch sw = Stopwatch()..start();
+  int exitCode = 1;
+  GradleHandledError detectedGradleError;
+  String detectedGradleErrorLine;
+  try {
+    exitCode = await processUtils.stream(
+      command,
+      workingDirectory: project.android.hostAppGradleRoot.path,
+      allowReentrantFlutter: true,
+      environment: gradleEnv,
+      mapFunction: (String line) {
+        // This message was removed from 1P plugins,
+        // but older plugin versions still display this message.
+        if (androidXPluginWarningRegex.hasMatch(line)) {
+          // Don't pipe.
+          return null;
+        }
+        if (detectedGradleError == null) {
+          for (final GradleHandledError gradleError in gradleErrors) {
+            if (gradleError.test(line)) {
+              detectedGradleErrorLine = line;
+              detectedGradleError = gradleError;
```

Comment:
Can you break out of the loop after a match?

Reply:
If you do that, make sure to document that the first one listed in the list wins.

Alternatively, collect all the matching errors and report all of them, or document that the last one listed in gradleErrors wins.

Reply:
Done